lake/sorted.go

```diff
@@ -78,8 +78,17 @@ func newSortedScanner(ctx context.Context, pool *Pool, zctx *zson.Context, filte
 			sched:   sched,
 		})
 	}
+	keys := pool.Layout.Keys
+	var merger zbuf.Puller
+	if len(pullers) == 1 {
+		merger = pullers[0]
+	} else if len(keys) > 0 && len(keys[0]) == 1 && keys[0][0] == "ts" {
+		merger = zbuf.MergeByTs(ctx, pullers, pool.Layout.Order)
```

Comment:
We're eventually going to have to address the fact that `zbuf.MergeByTs` doesn't do the right thing for records with a non-`time` `ts` field.

Reply:
I think this can easily be part of the JIT-CG.  The scan scheduler can look at the metadata across a partition to see inspect the type signature(s) for the pool key.